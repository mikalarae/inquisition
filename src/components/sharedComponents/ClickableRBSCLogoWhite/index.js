/** @jsx jsx */
// eslint-disable-next-line no-unused-vars
import React from 'react'
import PropTypes from 'prop-types'
import { jsx, Image } from 'theme-ui'
import ndLogo from '@ndlib/gatsby-theme-marble/src/assets/logos/rbsc-logo.svg'

const ClickableRBSCLogoWhite = ({ variant }) => {
  return (
    <div sx={{ variant: variant }}>
      <a href='https://rarebooks.library.nd.edu/' >
        <Image src={ndLogo} alt='Rare Books & Special Collections, Hesburgh Libraries, University of Notre Dame' />
      </a>
    </div>
  )
}

ClickableRBSCLogoWhite.propTpyes = {
  variant: PropTypes.string,
}

ClickableRBSCLogoWhite.defaultProps = {
  variant: 'ClickableRBSCLogoWhite.primary',
}

export default ClickableRBSCLogoWhite
