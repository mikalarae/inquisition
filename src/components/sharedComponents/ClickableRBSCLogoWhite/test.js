import React from 'react'
import renderer from 'react-test-renderer'
import ClickableRBSCLogoWhite from './'

describe('rbsc logo', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ClickableRBSCLogoWhite width={100} height={100} variant='variant' />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('sets width and heigh on the image', () => {
    const tree = renderer
      .create(<ClickableRBSCLogoWhite width={50} height={100} variant='variant' />)
      .toJSON()

    expect(tree.children[0].children[0].props.height).toEqual(100)
    expect(tree.children[0].children[0].props.width).toEqual(50)
    expect(tree.children[0].children[0].props.alt).toEqual('Rare Books & Special Collections, Hesburgh Libraries, University of Notre Dame')
  })
})
