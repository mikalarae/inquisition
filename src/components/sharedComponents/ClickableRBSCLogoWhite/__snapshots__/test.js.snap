// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rbsc logo renders correctly 1`] = `
<div
  className="css-udbkbz-ClickableRBSCLogoWhite"
>
  <a
    href="https://rarebooks.library.nd.edu/"
  >
    <img
      alt="Rare Books & Special Collections, Hesburgh Libraries, University of Notre Dame"
      className="css-9taffg"
      height={100}
      src="test-file-stub"
      width={100}
    />
  </a>
</div>
`;
